src/PackageVersions/FallbackVersions.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace PackageVersions;
+
+/**
+ * @internal
+ *
+ * This is a fallback for {@see \PackageVersions\Versions::getVersion()}
+ * Do not use this class directly: it is intended to be only used when
+ * {@see \PackageVersions\Versions} fails to be generated, which typically
+ * happens when running composer with `--no-scripts` flag)
+ */
+final class FallbackVersions
+{
+    private function __construct()
+    {
+    }
+
+    /**
+     * @param string $packageName
+     *
+     * @return string
+     *
+     * @throws \OutOfBoundsException if a version cannot be located
+     * @throws \UnexpectedValueException if the composer.lock file could not be located
+     */
+    public static function getVersion(string $packageName) : string
+    {
+        $versions = iterator_to_array(self::getVersions(self::getComposerLockPath()));
```

Comment:
copy-paste + laziness, mainly. Code can be improved, ofc, just didn't want to introduce bugs just by rewriting it